### PR TITLE
feat: add sftpsend plugin

### PR DESCRIPTION
**Pull request by matthieumarrast**

### CODE REVIEW

File: plugins/sftpsend/main.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                                                         default="FIXME")
+        if self.sftp_hostname == "FIXME" or self.sftp_hostname == "null" \
+                or self.sftp_hostname == "":
+            raise Exception("you have to set a valid sftp_hostname value")
+        if self.sftp_username == "FIXME" or self.sftp_username == "null" \
+                or self.sftp_username == "":
+            raise Exception("you have to set a valid sftp_username value")
+        if self.sftp_password == "FIXME" or self.sftp_password == "null" \
+                or self.sftp_password == "":
+            raise Exception("you have to set a valid sftp_password value")
```
Comment (thefab) [inline plugins/sftpsend/main.py:30]: I don't know SFTP very much but if you use a key_file, you don't need password, do you?

Comment (matthieumarrast) [reply plugins/sftpsend/main.py]: OK fixed

**Review by thefab — CHANGES_REQUESTED** [review]
we also need the requirements3.txt files

as we don't want any surprise if we rebuild the plugin two years later in a production branch if you see what I mean ;-)

**Comment by thebaptiste** [comment]
Two remarks : 
- for integration, master and release_1.1 branches, we probably don't need to force cryptography < 3 (for these branches, the cryptography version in mfext is 3.3.2). I think we probably don't need to have cryptography at all in requirement files because it's provided by mfext
- pysftp is a very old project which hasn't been released since July 2016, it seems that it has not been tested with python > 3.4 and it is probably not maintained anymore. May be a more recent sftp project would be better ? There are several "sftp" projects in pypi, such as sftp-py. But it's just a remark, may be pysftp is great.

**Comment by thefab** [comment]
yes @thebaptiste!

@matthieumarrast is going to push the virtualenv deletion
(it was here only because it's a port of a plugin developed for 0.9)

**Comment by matthieumarrast** [comment]
> 
> 
> yes @thebaptiste!
> 
> @matthieumarrast is going to push the virtualenv deletion
> (it was here only because it's a port of a plugin developed for 0.9)

DONE

**Comment by matthieumarrast** [comment]
> 
> 
> Two remarks :
> 
>     * for integration, master and release_1.1 branches, we probably don't need to force cryptography < 3 (for these branches, the cryptography version in mfext is 3.3.2). I think we probably don't need to have cryptography at all in requirement files because it's provided by mfext ( < 3 for release_1.0, > 3 for release >= 1.1)
> 
>     * pysftp is a very old project which hasn't been released since July 2016, it seems that it has not been tested with python > 3.4 and it is probably not maintained anymore. May be a more recent sftp project would be better ? There are several "sftp" projects in pypi, such as sftp-py. But it's just a remark, may be pysftp is great and there will probably never be a new version of pysftp, even in two years, so we don't risk any surprise !

Indeed I have removed `pysftp` and `cryptography`, replaced by `paramiko`.

**Review by thefab — APPROVED** [review]

**Pull request merged**